src/jailer/src/env.rs

```diff
@@ -575,17 +577,35 @@ impl Env {
                 let host_cache_file = host_path.join(entry);
                 let jailer_cache_file = jailer_path.join(entry);
 
-                let line = readln_special(&host_cache_file)?;
-                writeln_special(&jailer_cache_file, line)?;
+                if let Ok(line) = readln_special(&host_cache_file) {
+                    writeln_special(&jailer_cache_file, line)?;
+
+                    // We now change the permissions.
+                    let dest_path_cstr = to_cstring(&jailer_cache_file)?;
+                    // SAFETY: Safe because dest_path_cstr is null-terminated.
+                    SyscallReturnCode(unsafe {
+                        libc::chown(dest_path_cstr.as_ptr(), self.uid(), self.gid())
+                    })
+                    .into_empty_result()
+                    .map_err(|err| {
+                        JailerError::ChangeFileOwner(jailer_cache_file.to_owned(), err)
+                    })?;
+                } else {
+                    err_str += entry;
+                    err_str += ", ";
```

Comment:
Instead of building an error string you can just print the error inline:
```rust
eprintln!("Cache level {index}: {entry} entry not found, skipping");
```
This is a simpler approach, event thought it will produce more output.

Reply:
Does the jailer even have to print this at all if Firecracker is gonna print it later anyway?

Reply:
@ShadowCurse @roypat Happy to update for either case, I'll let you decide which you'd prefer. 

I'm assuming that if I updating for @roypat's version, the code should be updated to continue in the absence of a file, rather than blanket copying anything in `/sys/devices/system/cpu/cpuX/cache/indexY/` over?

Reply:
I was thinking we just drop all of the string manipulations and the `else` branch on the `if let`, and leave the rest as is

Reply:
Addressed in [426b027](https://github.com/firecracker-microvm/firecracker/pull/5062/commits/426b027d0d979fa238ce09214ae3f2571d48a899). The commit also contains a body, so it should pass the previously failing check.

Reply:
Thanks, looks good to me!